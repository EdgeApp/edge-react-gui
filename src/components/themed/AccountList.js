// @flow

import { type EdgeUserInfo } from 'edge-core-js'
import * as React from 'react'
import { Pressable, ScrollView, TouchableHighlight, View } from 'react-native'
import Animated, { Easing, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated'
import Feather from 'react-native-vector-icons/Feather'
import MaterialIcon from 'react-native-vector-icons/MaterialIcons'
import { sprintf } from 'sprintf-js'

import { deleteLocalAccount } from '../../actions/AccountActions.js'
import { logoutRequest } from '../../actions/LoginActions.js'
import { Fontello } from '../../assets/vector'
import s from '../../locales/strings'
import { useEffect, useState } from '../../types/reactHooks'
import { useDispatch, useSelector } from '../../types/reactRedux'
import { ButtonsModal } from '../modals/ButtonsModal.js'
import { Airship } from '../services/AirshipInstance.js'
import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext'
import { DividerLine } from '../themed/DividerLine'
import { EdgeText } from '../themed/EdgeText'

type Props = {
  onPress: () => void,
  isOpen: boolean,
  duration: number
}

export function AccountList(props: Props) {
  const { isOpen, onPress, duration = 500 } = props
  const activeUsername = useSelector(state => state.core.account.username)
  const context = useSelector(state => state.core.context)
  const dispatch = useDispatch()
  const theme = useTheme()
  const styles = getStyles(theme)

  // Maintain the list of usernames:
  const [usernames, setUsernames] = useState(arrangeUsers(context.localUsers, activeUsername))
  useEffect(() => context.watch('localUsers', localUsers => setUsernames(arrangeUsers(localUsers, activeUsername))), [activeUsername, context])

  const handleDelete = (username: string) => {
    Airship.show(bridge => (
      <ButtonsModal
        bridge={bridge}
        title={s.strings.delete_account_header}
        message={sprintf(s.strings.delete_username_account, username)}
        buttons={{
          ok: {
            label: s.strings.string_delete,
            async onPress() {
              await dispatch(deleteLocalAccount(username))
              height.value -= styles.row.height
              return true
            }
          },
          cancel: { label: s.strings.string_cancel }
        }}
      />
    ))
  }

  const handleSwitchAccount = (username: string) => {
    dispatch(logoutRequest(username))
  }

  const listHeight = styles.row.height * usernames.length + theme.rem(1)
  const height = useSharedValue(isOpen ? listHeight : 0)
  useEffect(() => {
    height.value = isOpen ? listHeight : 0
  }, [height, isOpen, listHeight])

  const animatedStyle = useAnimatedStyle(() => {
    return {
      height: withTiming(height.value, {
        duration: duration,
        easing: Easing.out(Easing.exp)
      })
    }
  })

  return (
    <View>
      <Pressable onPress={onPress}>
        <View style={styles.header}>
          <Fontello name="edge.logo" style={styles.iconUser} size={theme.rem(1.5)} color={theme.mainMenuIcon} />
          <EdgeText style={styles.text}>{activeUsername}</EdgeText>
          <Feather name="chevron-down" color={theme.mainMenuIcon} size={theme.rem(1.5)} />
        </View>
      </Pressable>
      <DividerLine marginRem={[1, -2, 0, 0]} />
      <Animated.View style={[styles.root, animatedStyle]}>
        <ScrollView>
          {usernames.map((username: string) => (
            <View key={username} style={styles.row}>
              <TouchableHighlight onPress={() => handleSwitchAccount(username)}>
                <EdgeText style={styles.text}>{username}</EdgeText>
              </TouchableHighlight>
              <TouchableHighlight onPress={() => handleDelete(username)}>
                <View>
                  <MaterialIcon size={theme.rem(1.5)} name="close" color={theme.mainMenuIcon} />
                </View>
              </TouchableHighlight>
            </View>
          ))}
        </ScrollView>
      </Animated.View>
    </View>
  )
}

/**
 * Given a list of users from the core,
 * remove the given user, then organize the 3 most recent users,
 * followed by the rest in alphabetical order.
 */
function arrangeUsers(localUsers: EdgeUserInfo[], activeUsername: string): string[] {
  // Sort the users according to their last login date:
  const usernames = localUsers
    .filter(info => info.username !== activeUsername)
    .sort((a, b) => {
      const { lastLogin: aDate = new Date(0) } = a
      const { lastLogin: bDate = new Date(0) } = b
      return aDate.valueOf() - bDate.valueOf()
    })
    .map(info => info.username)

  // Sort everything after the last 3 entries alphabetically:
  const oldUsernames = usernames.slice(3).sort((a: string, b: string) => {
    const stringA = a.toUpperCase()
    const stringB = b.toUpperCase()
    if (stringA < stringB) return -1
    if (stringA > stringB) return 1
    return 0
  })

  return [...usernames.slice(0, 3), ...oldUsernames]
}

const getStyles = cacheStyles((theme: Theme) => ({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    height: theme.rem(2.5),
    marginLeft: theme.rem(2)
  },
  text: {
    marginRight: 'auto',
    fontFamily: theme.fontFaceBold
  },
  header: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    height: theme.rem(2)
  },
  iconUser: {
    marginRight: theme.rem(1.5)
  }
}))
