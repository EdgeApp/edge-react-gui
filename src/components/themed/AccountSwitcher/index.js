// @flow

export * from './helpers'
export { default as SwitcherHeader } from './SwitcherHeader'
export { default as SwitcherList } from './SwitcherList'
