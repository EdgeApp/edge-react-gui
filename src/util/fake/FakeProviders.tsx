import * as React from 'react'
import { Metrics, SafeAreaProvider } from 'react-native-safe-area-context'
import { Provider } from 'react-redux'
import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'

import { rootReducer, RootState } from '../../reducers/RootReducer'

type DeepPartial<T> = T extends object
  ? {
      [P in keyof T]?: DeepPartial<T[P]>
    }
  : T

export type FakeState = DeepPartial<RootState>

interface Props {
  children: React.ReactNode
  initialState?: FakeState
}

export function FakeProviders(props: Props) {
  const { children, initialState = {} } = props

  const store = React.useMemo(() => createStore(rootReducer, initialState as any, applyMiddleware(thunk)), [initialState])
  return (
    <SafeAreaProvider initialMetrics={initialMetrics}>
      <Provider store={store}>{children}</Provider>
    </SafeAreaProvider>
  )
}

// These match what our old react-native-safe-area-view library returned:
const initialMetrics: Metrics = {
  frame: { height: 1334, width: 750, x: 0, y: 0 },
  insets: { bottom: 0, left: 0, right: 0, top: 20 }
}
