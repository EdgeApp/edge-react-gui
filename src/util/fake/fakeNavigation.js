// @flow

import { type NavigationProp } from '../../types/routerTypes.js'

export const fakeNavigation: NavigationProp<any> = {
  addListener(event, callback) {
    return () => undefined
  },
  isFocused() {
    return true
  },

  navigate(name, params) {},
  push(name, params) {},
  replace(name, params) {},
  setParams(params) {},
  jumpTo(name, params) {},

  goBack() {},
  pop() {},
  popToTop() {},

  closeDrawer() {},
  openDrawer() {},
  toggleDrawer() {},

  state: {},
  currentScene: {}
}
