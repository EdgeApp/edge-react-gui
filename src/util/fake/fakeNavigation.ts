import { NavigationProp } from '../../types/routerTypes'

export const fakeNavigation: NavigationProp<any> = {
  addListener(event, callback) {
    return () => undefined
  },
  isFocused() {
    return true
  },

  navigate(name, params) {},
  push(name, params) {},
  replace(name, params) {},
  setParams(params) {},

  goBack() {},
  pop() {},
  popToTop() {},

  closeDrawer() {},
  openDrawer() {},
  toggleDrawer() {},

  state: {}
}
