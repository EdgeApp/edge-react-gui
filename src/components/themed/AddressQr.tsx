import { EdgeCurrencyWallet, EdgeTokenId } from 'edge-core-js'
import * as React from 'react'
import { View } from 'react-native'

import { useAsyncValue } from '../../hooks/useAsyncValue'
import { getCurrencyCode } from '../../util/CurrencyInfoHelpers'
import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
import { QrCode } from '../themed/QrCode'

interface Props {
  address: string
  tokenId: EdgeTokenId
  wallet: EdgeCurrencyWallet

  nativeAmount?: string
  onPress?: (encodedUri?: string) => void
}

export const AddressQr = (props: Props) => {
  const theme = useTheme()
  const styles = getStyles(theme)
  const { address, tokenId, wallet, nativeAmount, onPress = () => {} } = props

  const [encodedUri] = useAsyncValue(
    async () => await wallet.encodeUri({ publicAddress: address, currencyCode: getCurrencyCode(wallet, tokenId), nativeAmount }),
    [address, tokenId, nativeAmount, wallet]
  )

  return (
    <View style={styles.container}>
      <QrCode data={encodedUri} tokenId={tokenId} wallet={wallet} onPress={() => onPress(encodedUri)} marginRem={0} />
    </View>
  )
}

const getStyles = cacheStyles((theme: Theme) => ({
  container: {
    aspectRatio: 1,
    height: '100%',
    width: '100%',
    alignSelf: 'center'
  }
}))
