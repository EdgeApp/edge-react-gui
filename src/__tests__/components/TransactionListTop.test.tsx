import { describe, expect, it } from '@jest/globals'
import { EdgeCurrencyInfo } from 'edge-core-js'
import * as React from 'react'
import TestRenderer from 'react-test-renderer'

import { TransactionListTop } from '../../components/themed/TransactionListTop'
import { fakeNavigation } from '../../util/fake/fakeNavigation'
import { FakeProviders, FakeState } from '../../util/fake/FakeProviders'

describe('TransactionListTop', () => {
  const currencyInfo: EdgeCurrencyInfo = {
    pluginId: 'bitcoin',
    currencyCode: 'BTC',
    displayName: 'Bitcoin',
    walletType: 'bitcoin',

    addressExplorer: '',
    transactionExplorer: '',

    defaultSettings: {},
    metaTokens: [],
    denominations: [
      {
        name: 'BTC',
        multiplier: '100000000',
        symbol: 'â‚¿'
      }
    ]
  }

  const fakeCurrencyConfig: any = {
    currencyInfo,
    allTokens: {},
    builtinTokens: {},
    customTokens: {}
  }

  const fakeWallet: any = {
    balances: { BTC: '123123' },
    blockHeight: 12345,
    currencyConfig: fakeCurrencyConfig,
    currencyInfo,
    enabledTokenIds: [],
    fiatCurrencyCode: 'USD',
    id: '123',
    name: 'wallet name',
    type: 'wallet:bitcoin',
    watch() {}
  }

  const fakeState: FakeState = {
    core: {
      account: {
        currencyWallets: { '123': fakeWallet },
        currencyConfig: { bitcoin: fakeCurrencyConfig }
      }
    }
  }

  it('should render', () => {
    const renderer = TestRenderer.create(
      <FakeProviders initialState={fakeState}>
        <TransactionListTop
          currencyCode="BTC"
          isEmpty={false}
          navigation={fakeNavigation}
          searching={false}
          wallet={fakeWallet}
          onChangeSortingState={() => undefined}
          onSearchTransaction={() => undefined}
        />
      </FakeProviders>
    )

    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
