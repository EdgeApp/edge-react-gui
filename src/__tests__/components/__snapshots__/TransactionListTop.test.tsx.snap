// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionListTop should render 1`] = `
Array [
  <View
    style={
      Object {
        "flex": 1,
        "marginBottom": 11,
        "paddingBottom": 22,
        "paddingHorizontal": 22,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 11,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "rgba(255, 255, 255, 0)",
                "borderRadius": 11,
                "justifyContent": "center",
                "minHeight": 67,
                "paddingHorizontal": 22,
              },
              Object {
                "flexGrow": 0,
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "paddingLeft": 65,
                "paddingRight": 65,
                "paddingVertical": 0,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderTopWidth": 1,
                  "bottom": 0,
                  "left": 22,
                  "position": "absolute",
                  "right": 22,
                },
                Object {
                  "borderColor": undefined,
                  "right": 11,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "borderBottomLeftRadius": 11,
                  "borderBottomWidth": 1,
                  "borderLeftWidth": 1,
                  "borderRightWidth": 0,
                  "borderTopLeftRadius": 11,
                  "borderTopWidth": 1,
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 22,
                },
                Object {
                  "borderColor": undefined,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "borderBottomRightRadius": 11,
                  "borderBottomWidth": 1,
                  "borderLeftWidth": 0,
                  "borderRightWidth": 1,
                  "borderTopRightRadius": 11,
                  "borderTopWidth": 1,
                  "bottom": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 22,
                },
                Object {
                  "borderColor": undefined,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "borderTopWidth": 1,
                  "left": 22,
                  "position": "absolute",
                  "right": 22,
                  "top": 0,
                },
                Object {
                  "borderColor": undefined,
                  "left": 22,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "height": 67,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingHorizontal": 22,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "paddingLeft": 65,
                  "paddingRight": 65,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              onLayout={[Function]}
              style={
                Array [
                  Object {
                    "alignSelf": "flex-start",
                    "fontFamily": "Quicksand-Regular",
                    "fontSize": 22,
                    "padding": 0,
                  },
                  Object {
                    "color": undefined,
                    "transform": Array [
                      Object {
                        "translateX": -0,
                      },
                      Object {
                        "translateY": -0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                  },
                ]
              }
            >
              Search Transactions
            </Text>
          </View>
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "bottom": -28,
                  "color": "#E85466",
                  "fontFamily": "Quicksand-Regular",
                  "fontSize": 17,
                  "left": 28,
                  "position": "absolute",
                },
                Object {
                  "opacity": 0,
                },
              ]
            }
          />
          <Text
            numberOfLines={1}
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "bottom": -10,
                  "fontFamily": "Quicksand-Regular",
                  "fontSize": 17,
                  "position": "absolute",
                  "right": 28,
                },
                Object {
                  "color": undefined,
                  "opacity": 0,
                  "transform": Array [
                    Object {
                      "translateX": 6,
                    },
                    Object {
                      "scale": 0,
                    },
                  ],
                },
              ]
            }
          >
            
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "rgba(255, 255, 255, .75)",
                  "fontSize": 22,
                  "left": 0,
                  "padding": 22,
                  "position": "absolute",
                  "top": 0,
                },
                Object {
                  "fontFamily": "anticon",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            î™°
          </Text>
          <TextInput
            autoFocus={false}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            returnKeyType="search"
            secureTextEntry={false}
            selectionColor="#FFFFFF"
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "color": "#FFFFFF",
                  "fontFamily": "Quicksand-Regular",
                  "fontSize": 22,
                  "padding": 0,
                },
                Object {
                  "flexGrow": 0,
                },
              ]
            }
            textAlignVertical="top"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 34,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "flexShrink": 1,
                  "marginRight": 11,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 34,
                    "marginBottom": 0,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 0,
                    "width": 34,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={50}
                  bbWidth={50}
                  focusable={false}
                  height={50}
                  meetOrSlice={0}
                  minX={-8}
                  minY={-8}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "left": -8,
                        "position": "absolute",
                        "top": -8,
                        "transform": Array [
                          Object {
                            "rotateZ": "-90deg",
                          },
                        ],
                      },
                      Object {
                        "flex": 0,
                        "height": 50,
                        "width": 50,
                      },
                    ]
                  }
                  vbHeight={50}
                  vbWidth={50}
                  width={50}
                >
                  <RNSVGGroup
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                  >
                    <RNSVGCircle
                      cx={17}
                      cy={17}
                      propList={
                        Array [
                          "strokeWidth",
                          "strokeDasharray",
                          "strokeLinecap",
                        ]
                      }
                      r={19}
                      strokeDasharray={
                        Array [
                          119.38052083641213,
                          119.38052083641213,
                        ]
                      }
                      strokeLinecap={1}
                      strokeWidth={4}
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
                <View
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      },
                    ]
                  }
                >
                  <FastImageView
                    resizeMode="cover"
                    source={
                      Object {
                        "uri": "https://content.edge.app/currencyIconsV3/bitcoin/bitcoin.png",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  />
                </View>
              </View>
              <Text
                adjustsFontSizeToFit={true}
                minimumFontScale={0.65}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Quicksand-Regular",
                      "fontSize": 22,
                      "includeFontPadding": false,
                    },
                    Object {
                      "flexShrink": 1,
                      "fontSize": 28,
                      "marginLeft": 11,
                      "marginRight": 6,
                    },
                    null,
                  ]
                }
              >
                wallet name
              </Text>
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#00f1a2",
                      "fontSize": 34,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ïˆ”
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#00f1a2",
                      "fontSize": 34,
                    },
                    undefined,
                    Object {
                      "fontFamily": "custom",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                î ’
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              minimumFontScale={0.3}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Quicksand-Regular",
                    "fontSize": 22,
                    "includeFontPadding": false,
                  },
                  Object {
                    "fontFamily": "Quicksand-Medium",
                    "fontSize": 45,
                  },
                  null,
                ]
              }
            >
              0.00123123 BTC
            </Text>
            <Text
              adjustsFontSizeToFit={true}
              minimumFontScale={0.65}
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Quicksand-Regular",
                    "fontSize": 22,
                    "includeFontPadding": false,
                  },
                  Object {
                    "fontSize": 28,
                  },
                  null,
                ]
              }
            >
              $0.00 USD
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginTop": 22,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexShrink": 1,
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingRight": 34,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#00f1a2",
                "fontSize": 22,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î˜™
        </Text>
        <Text
          adjustsFontSizeToFit={true}
          minimumFontScale={0.65}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 22,
                "includeFontPadding": false,
              },
              Object {
                "color": "#00f1a2",
                "fontFamily": "Quicksand-Medium",
                "fontSize": 22,
                "marginLeft": 6,
              },
              null,
            ]
          }
        >
          Receive
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "flexShrink": 1,
            "justifyContent": "flex-start",
            "opacity": 1,
            "paddingRight": 34,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#00f1a2",
                "fontSize": 22,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î˜š
        </Text>
        <Text
          adjustsFontSizeToFit={true}
          minimumFontScale={0.65}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Quicksand-Regular",
                "fontSize": 22,
                "includeFontPadding": false,
              },
              Object {
                "color": "#00f1a2",
                "fontFamily": "Quicksand-Medium",
                "fontSize": 22,
                "marginLeft": 6,
              },
              null,
            ]
          }
        >
          Send
        </Text>
      </View>
      <ActivityIndicator
        color="#00f1a2"
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingRight": 22,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "rgba(255, 255, 255, .1)",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            null,
            Object {
              "marginBottom": -11,
              "marginLeft": 11,
              "marginRight": 11,
              "marginTop": 22,
            },
            Object {
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "padding": 11,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "height": 45,
                  "margin": 11,
                  "width": 45,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="contain"
              source={
                Object {
                  "uri": "https://content.edge.app/currencyIconsV3/bitcoin/bitcoin.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </View>
          <Text
            adjustsFontSizeToFit={true}
            minimumFontScale={0.65}
            numberOfLines={0}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "Quicksand-Regular",
                  "fontSize": 22,
                  "includeFontPadding": false,
                },
                Object {
                  "marginLeft": 11,
                },
                null,
              ]
            }
          >
            Spend USD â†’ Earn BTC
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "justifyContent": "center",
          "marginHorizontal": 22,
          "paddingBottom": 22,
        },
        null,
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginRight": 22,
        }
      }
    />
    <Text
      adjustsFontSizeToFit={true}
      minimumFontScale={0.65}
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Quicksand-Regular",
            "fontSize": 22,
            "includeFontPadding": false,
          },
          Object {
            "fontFamily": "Quicksand-Medium",
          },
          null,
        ]
      }
    >
      Transactions
    </Text>
  </View>,
  <BVLinearGradient
    colors={
      Array [
        452984831,
        452984831,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0.5,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0.5,
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "height": 1,
        },
        Object {
          "marginBottom": 0,
          "marginLeft": 22,
          "marginRight": 0,
          "marginTop": 0,
        },
      ]
    }
  />,
]
`;
